powder_reduction.py

```diff
@@ -23,111 +44,459 @@ def powder_reduction(sample='sample.nxs', calibration=None, lambda_binning=(0.7,
+              https://docs.mantidproject.org/nightly/algorithms/CylinderAbsorption-v1.html
+
+    """
+    vana_red = process_event_data(vanadium, lambda_binning)
+    ec_red = process_event_data(empty_instr, lambda_binning)
+
+    # vana - EC
+    vana_red -= ec_red
+
+    del ec_red
+
+    # Absorption
+    # TODO: adapt for different number of inputs
+    if bool(absorp):
+        correction = absorption_correction(vanadium,
+                                           lambda_binning,
+                                           ScatterFrom='Sample',
+                                           SampleNumberDensity=absorp['number_density'],
+                                           ScatteringXSection=absorp['scattering'],
+                                           AttenuationXSection=absorp['attenuation'],
+                                           CylinderSampleHeight=absorp['height'],
+                                           CylinderSampleRadius=absorp['radius'])
+    else:
+        correction = absorption_correction(vanadium,
+                                           lambda_binning,
+                                           ScatterFrom='Sample')
+
+    # the 5 following lines of code are to place info about source and sample
+    # position at the right place in the correction dataArray in order to
+    # proceed to the normalization
+
+    correction.attrs['source_position'] = vana_red.attrs['source_position'].copy()
+    correction.attrs['sample_position'] = vana_red.attrs['sample_position'].copy()
+
+    del correction.coords['source_position']
+    del correction.coords['sample_position']
+    del correction.coords['position']
+
+    correction = sc.rebin(correction,
+                          'wavelength',
+                          sc.Variable(['wavelength'],
+                                      values=vana_red.coords['wavelength'].values,
+                                      unit=sc.units.angstrom))
+
+    vana_red /= correction
+
+    del correction
+
+    # convert to TOF
+    vana_red_tof = sc.neutron.convert(vana_red, 'wavelength', 'tof')
+
+    del vana_red
+
+    # convert to d-spacing (no calibration applied)
+    vana_dspacing = sc.neutron.convert(vana_red_tof, 'tof', 'd-spacing')
+
+    del vana_red_tof
+
+    # rebin vanadium
+    dmin, dmax, number_bins_d = dspacing_binning
     dspacing_bins = sc.Variable(['d-spacing'],
-                                values=np.arange(1., 10., 0.001),
+                                values=np.linspace(dmin, dmax, number_bins_d),
                                 unit=sc.units.angstrom)
+    vana_rebin = sc.rebin(vana_dspacing, 'd-spacing', dspacing_bins)
 
-    focused_hist = sc.histogram(focused, dspacing_bins)
+    del vana_dspacing
 
-    del focused
+    # Calibration
+    # Load
+    input_load_cal = {'InstrumentFilename': 'WISH_Definition.xml'}
+    calvana = load_calibration(calibration, mantid_args=input_load_cal)
```

Comment:
Does this work properly now? When I ran this last week Mantid was really slow and had apparently trouble with tons of bad detector IDs.